msal/src/main/java/com/microsoft/identity/client/PublicClientApplicationConfiguration.java

```diff
@@ -610,4 +623,32 @@ public void checkIntentFilterAddedToAppManifestForBrokerFlow() throws MsalClient
         verifyRedirectUriWithAppSignature();
     }
 
+    private boolean isValidAuthenticatorRedirectUri() {
```

Comment:
+1

Reply:
I think what I was trying to say was.... 

1. Apart from the oob check, the rest of the logic is identical.
2. if we add an early-return check like i did, we could just remove that line.

Reply:
Early return also makes it easier to see the deltas, IMO.

Reply:
We still need to verify the package name and signature before returning early, right
So it is not that early :).. 
Let me know if you think the suggested change is must?

Reply:
We don't. just move line 643 /w Package name check all the way up.

if (package name = authApp AND sig = the old broker one){
 return true
}

Reply:
(In general, i'm a fan of less code, if possible - it's easier to fix/read for other engineers who are not involved

In this case, i think it's a small change, so we should do it.)

Reply:
That will force us to also update the order from line 610-620. Otherwise the whole logic will fail.
I can make the change but I don't think it's necessary as we are going to remove this code anyway, 
So I am trying to avoid modifying anything that is existing and just adding additional checks for Authenticator.